x-pack/plugins/transform/public/app/sections/create_transform/components/step_define/common/top_metrics_agg/config.ts

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import {
+  isPivotAggsConfigWithUiSupport,
+  isSpecialSortField,
+  isValidSortDirection,
+  isValidSortMode,
+  isValidSortNumericType,
+  PivotAggsConfigBase,
+  PivotAggsConfigWithUiBase,
+} from '../../../../../../common/pivot_aggs';
+import { PivotAggsConfigTopMetrics } from './types';
+import { TopMetricsAggForm } from './components/top_metrics_agg_form';
+
+/**
+ * Gets initial basic configuration of the top_metrics aggregation.
+ */
+export function getTopMetricsAggConfig(
+  commonConfig: PivotAggsConfigWithUiBase | PivotAggsConfigBase
+): PivotAggsConfigTopMetrics {
+  return {
+    ...commonConfig,
+    isSubAggsSupported: false,
+    isMultiField: true,
+    field: isPivotAggsConfigWithUiSupport(commonConfig) ? commonConfig.field : '',
+    AggFormComponent: TopMetricsAggForm,
+    aggConfig: {},
+    getEsAggConfig() {
+      // ensure the configuration has been completed
+      if (!this.isValid()) {
+        return null;
+      }
+
+      const {
+        sortField,
+        sortDirection,
+        sortMode,
+        numericType,
+        ...unsupportedConfig
+      } = this.aggConfig;
+
+      let sort = null;
+
+      if (isSpecialSortField(sortField)) {
+        sort = sortField;
+      } else {
+        if (sortMode || numericType) {
+          sort = {
+            [sortField!]: {
+              order: sortDirection,
+              ...(sortMode ? { mode: sortMode } : {}),
+              ...(numericType ? { numeric_type: numericType } : {}),
+              ...(unsupportedConfig ?? {}),
+            },
+          };
+        } else {
+          sort = { [sortField!]: sortDirection };
+        }
+      }
+
+      return {
+        metrics: (Array.isArray(this.field) ? this.field : [this.field]).map((f) => ({ field: f })),
+        sort,
+      };
+    },
+    setUiConfigFromEs(esAggDefinition) {
+      const { metrics, sort } = esAggDefinition;
+
+      this.field = (Array.isArray(metrics) ? metrics : [metrics]).map((v) => v.field);
+
+      if (isSpecialSortField(sort)) {
+        this.aggConfig.sortField = sort;
+        return;
+      }
+
+      const sortField = Object.keys(sort)[0];
+      const sortDefinition = sort[sortField];
+
+      let sortDirection = null;
+      let sortMode = null;
+      let numericType = null;
+      let unsupportedConfig = null;
+
+      if (isValidSortDirection(sortDefinition)) {
+        sortDirection = sortDefinition;
+      }
+
+      if (typeof sortDefinition === 'object') {
```

Comment:
nit: I think we can use isPopulatedObject here

Reply:
Updated in https://github.com/elastic/kibana/pull/101152/commits/24b7c6f2dcadb1ce3a5d7a823b609dfd0e197be4